### PR TITLE
Cleanup docs, docstrings, and setup.py metadata.

### PR DESCRIPTION
**Pull request by jcsackett**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/88/
Test PASSed.

File: README.rst [hunk 1]
```diff
@@ -10,14 +10,20 @@ The easiest way to install theblues is via pip::
 
     $ pip install theblues
 
-Note that theblues requires python-macaroons (which has its own dependencies),
-which must be installed from a ppa::
+Parts of theblues requires use of macaroons (e.g. parsing the return of
```
Comment (bac) [inline README.rst]: typo: require the use

File: README.rst [hunk 2]
```diff
@@ -10,14 +10,20 @@ The easiest way to install theblues is via pip::
...
+manage macaroons. theblues was developed around libmacaroons. On ubuntu, you
+can get libmacaroons from a ppa::
 
 
 	$ sudo add-apt-repository ppa:yellow/ppa -y
 	$ apt-get install libmacaroons0 python-macaroons libsodium13
 
-Without these, theblues cannot communicate with the charmstore.
+Without these, theblues cannot make any authenticated requests to the
+charmstore or other services, but is usable to communicate with charmstore for
```
Comment (bac) [inline README.rst]: the charmstore

File: theblues/charmstore.py [hunk 3]
```diff
@@ -21,13 +21,28 @@ class CharmStore(object):
     """A connection to the charmstore."""
 
     def __init__(self, url, macaroons=None, timeout=None, verify=True):
+        """Initializer.
+
+        @param url The url to the charmstore API.
+        @param timeout How long to wait before timing out a request; a value of
+            None means no timeout.
```
Comment (bac) [inline theblues/charmstore.py]: I assume timeout is in seconds. Please state it.

File: README.rst (hunk 2 above)
Comment (frankban) [inline README.rst]: the charm store?

File: theblues/charmstore.py [hunk 4]
```diff
@@ -21,13 +21,28 @@ class CharmStore(object):
     """A connection to the charmstore."""
 
     def __init__(self, url, macaroons=None, timeout=None, verify=True):
+        """Initializer.
+
+        @param url The url to the charmstore API.
+        @param timeout How long to wait before timing out a request; a value of
```
Comment (frankban) [inline theblues/charmstore.py]: missing macaroon param.

File: theblues/charmstore.py [hunk 5]
```diff
@@ -21,13 +21,28 @@ class CharmStore(object):
...
         super(CharmStore, self).__init__()
         self.url = url
         self.verify = verify
         self.timeout = timeout
         self.macaroons = macaroons
 
     def _get(self, url):
+        """Make a get request against the charmstore.
+
+        This method is used by other api methods to standardize querying.
```
Comment (frankban) [inline theblues/charmstore.py]: s/api/API/

File: theblues/jem.py [hunk 6]
```diff
@@ -47,22 +51,20 @@ def fetch_macaroon(self):
...
+        @return The json decoded list of environments.
         """
         return make_request("{}env".format(self.url), macaroons=macaroons)
 
     def get_model(self, macaroons, user, name):
         """ Get a specified model.
 
         @param macaroons The discharged JEM macaroons.
-        @param user The username.
+        @param user The model's owner's username.
```
Comment (bac) [inline theblues/jem.py]: The double possessives are hard to read.  Wouldn't "The model owner's username" say the same thing?

Or "The username of the model's owner"

**Comment by bac** [comment]
👍 with tweaks

File: theblues/charmstore.py [hunk 7]
```diff
@@ -361,6 +401,10 @@ def _common_query_parameters(self, doc_type, includes, owner,
         return queries
 
     def fetch_related(self, ids):
+        """Fetch related entity information.
+
+        @param ids The entity ids to fetch related information for.
```
Comment (frankban) [inline theblues/charmstore.py]: ... like stats, extra information, bundle metadata, and unit count for bundle identifiers, supported series for charms...
or similar?

Comment (frankban) [inline theblues/charmstore.py]: Also, this params really does not explain what type is expected here. I thought it was just an iterator of strings, but looking at the code it does not seem true.

File: theblues/charmstore.py [hunk 8]
```diff
@@ -373,10 +417,10 @@ def fetch_related(self, ids):
         return data.json().values()
 
     def fetch_interfaces(self, interface, way):
-        """Get the list of charms that provides or requires this id
+        """Get the list of charms that provides or requires this interface.
 
-        @param id: charm string
-        @param way: provides or requires
+        @param interface The interface for the charm relation.
+        @param way The type of relation, either provides or requires.
```
Comment (frankban) [inline theblues/charmstore.py]: either "provides" or "requires"?

File: theblues/identity_manager.py [hunk 9]
```diff
@@ -20,16 +20,24 @@ class IdentityManager(object):
     """Identity Manager API."""
 
     def __init__(self, url, idm_user, idm_password, timeout=3.05):
+        """Initializer.
+
+        @param url The url to the identity manager (IdM) API.
+        @param idm_user The user name for the IdM.
+        @param idm_password The password for the IdM.
+        @param timeout How long to wait before timing out a request; a value of
```
Comment (frankban) [inline theblues/identity_manager.py]: in seconds

File: theblues/identity_manager.py [hunk 10]
```diff
@@ -61,11 +69,11 @@ def login(self, username, json_document):
...
         """Discharge the macarooon for the identity.
 
-        @param username -- the logged in user.
-        @param macaroon -- the macaroon returned from CharmStore.
-
-        Return the resulting base64 encoded macaroon.
         Raise a ServerError if an error occurs in the request process.
+
+        @param username The logged in user.
+        @param macaroon The macaroon returned from the Charmstore.
```
Comment (frankban) [inline theblues/identity_manager.py]: I'd use "charm store" everywhere, but that's just personal preference. Anyway, let's use the same spelling everywhere in the code.

**Comment by frankban** [comment]
👍 with only minor comments. Great cleanup thank you!

**Comment by jcsackett** [comment]
Thanks for the comments, @frankban and @bac. I'll fix those and land this.

**Comment by jcsackett** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/89/
Test PASSed.

**Pull request merged**